src/main.ts

```diff
@@ -228,6 +230,56 @@ export function generateFile(typeMap: TypeMap, fileDesc: FileDescriptorProto, pa
     file = addTimestampMethods(file, options);
   }
 
+  if (options.outputMetaTypings) {
```

Comment:
Just thinking ahead (...and also in retrospect), `main.ts`  is already too big, can you move the contents of this `if` to a new `generate-metadata.ts` (or `generate-schema.ts`)?